wdl2cwl/main_oop.py

```diff
@@ -205,20 +253,41 @@ def get_expr_apply(self, wdl_apply_expr: WDL.Expr.Apply) -> str:
         treat_as_optional = wdl_apply_expr.type.optional
         if function_name == "_add":
             left_operand, right_operand = arguments
-            right_operand = self.get_wdl_literal(right_operand.literal)  # type: ignore
+            if isinstance(right_operand, WDL.Expr.String):
+                right_operand = self.get_expr_string(right_operand)  # type: ignore
+            else:
+                right_operand = self.get_wdl_literal(right_operand.literal)  # type: ignore
             left_operand_value = self.get_expr(left_operand)
             if getattr(left_operand, "function_name", None) == "basename":
                 treat_as_optional = True
                 referer = wdl_apply_expr.parent.name  # type: ignore
+            # return_str = f"{left_operand_value} + {right_operand}" if not isinstance(type(right_operand), str) else f"{left_operand_value} + '{right_operand}'"
             return (
                 f"{left_operand_value} + {right_operand}"
                 if not treat_as_optional
-                else f"{self.get_input(referer)} === null ? {left_operand_value} + '{right_operand}' : {self.get_input(referer)}"
+                else f"{self.get_input(referer)} === null ? {left_operand_value} + {right_operand} : {self.get_input(referer)}"
             )
         elif function_name == "basename":
-            only_operand = arguments[0]
-            only_operand = self.get_expr_name(only_operand.expr)  # type: ignore
-            return f"{only_operand}.basename"
+            if len(arguments) == 1:
+                only_operand = arguments[0]
+                is_file = isinstance(only_operand.type, WDL.Type.File)
+                only_operand = self.get_expr_name(only_operand.expr)  # type: ignore
+                return (
+                    f"{only_operand}.basename"
+                    if is_file
+                    else f"{only_operand}.split('/').reverse()[0]"
+                )
+            elif len(arguments) == 2:
+                operand, suffix = arguments
+                is_file = isinstance(operand.type, WDL.Type.File)
+                operand = self.get_expr_name(operand.expr)  # type: ignore
+                suffix_str = self.get_wdl_literal(suffix.literal)  # type: ignore
+                regex_str = suffix_str
```

Comment:
What happened to the `re.escape`?

Reply:
Since we are using "/". Javascript has a different escape character I guess.

Reply:
Those slashes are JS syntax that inside them is a regular expression. We still need to escape the contents. Otherwise any periods with be interpreted as "any character" instead of "must be the period character", for example

Reply:
you are totally right. I just tested it. Awesome.